logging/monitor/monitor.go

```diff
@@ -35,7 +35,7 @@ type monitor struct {
 	droppedCount int
 
 	// doneCh coordinates the shutdown of logCh
-	doneCh chan struct{}
+	doneCh chan bool
```

Comment:
Your question over in https://github.com/hashicorp/consul/pull/9285#discussion_r650157725 made me think of this. I think this is actually not safe for the reason I outline there.

This needs to be a broadcast, because multiple goroutines need to be informed of the change (both the `Start` goroutine, and   whatever is calling `Write`). 

If we need to change a bunch of tests to account for the flushing change let me know, maybe I can help with that.

Reply:
I reverted the change, anyway after reading the code closely it's not meant to be concurrent (we create a new monitor in the handler for each request, reusing the same monitor will not work on both start and stop)